### ISSUE TITLE
CORS issue preventing use of url_launcher

### ISSUE THREAD

**Issue by RedBrogdon** [dart-lang/dart-pad#2069]
While some websites can be opened from DartPad with the url_launcher package, some fail to load and render correctly due to sandbox-related errors:

This gist can be used for testing: https://dartpad.dev/?id=d8469da92364a32613dabf9a52681080

Launching https://youtube.com/flutterdev results in this error:

```
Uncaught DOMException: Failed to read the 'cookie' property from 'Document': The document is sandboxed and lacks the 'allow-same-origin' flag.
```

And launching https://dart.dev gets these:

```
Access to CSS stylesheet at 'https://dart.dev/assets/main-65831f33f424ced2a1a8fc47cae76ac0a9a4db598e95840b7fd015ef4f99ba98.css' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
main-65831f33f424ced2a1a8fc47cae76ac0a9a4db598e95840b7fd015ef4f99ba98.css:1 Failed to load resource: net::ERR_FAILED
dart.dev/:1 Access to script at 'https://dart.dev/assets/main-1058793e0683fd1820bc36e8b2f3d939c23110503201c4addf5e26a583bcb853.js' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
main-1058793e0683fd1820bc36e8b2f3d939c23110503201c4addf5e26a583bcb853.js:1 Failed to load resource: net::ERR_FAILED
content_script_compiled.js:32 Uncaught SecurityError: Sandbox access violation: Blocked a frame at "null" from accessing a frame at "https://dartpad.dev".  Both frames are sandboxed and lack the "allow-same-origin" flag.
(anonymous) @ content_script_compiled.js:32
embed_dart.dart.js:671 Uncaught Error: Failed to read the 'localStorage' property from 'Window': The document is sandboxed and lacks the 'allow-same-origin' flag.
```

**Comment by domesticmouse** [comment]
I'd lean towards documenting the restriction over loosening the sandbox.

**Comment by srawlins** [comment]
Yeah I think the discussion in https://github.com/dart-lang/dart-pad/pull/2044 showed that putting the `allow-same-origin` sandbox property on the iframe would be too great a security gap. I'll have to consider other options but I am not confident there will be another solution.

We might be able to programatically set `allow-same-origin` depending on the user script, like based on `import dart:js` or something, but I think that is really setting up a swiss cheese policy...

**Comment by parlough** [comment]
The URL launching use cases seem to be working properly now, as can be seen in the [Google AI SDK sample](https://dartpad.dev/?sample=google-ai-sdk).

If anyone is still experiencing issues with `package:url_launcher`, please open a new issue with reproduction details. Thanks so much!
